src/main/java/org/truffleruby/core/cast/BigIntegerCastNode.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2013, 2020 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 2.0, or
+ * GNU General Public License version 2, or
+ * GNU Lesser General Public License version 2.1.
+ */
+package org.truffleruby.core.cast;
+
+import java.math.BigInteger;
+
+import org.truffleruby.Layouts;
+import org.truffleruby.RubyContext;
+import org.truffleruby.RubyLanguage;
+import org.truffleruby.language.RubySourceNode;
+import org.truffleruby.language.RubyGuards;
+import org.truffleruby.language.RubyNode;
+import org.truffleruby.language.control.RaiseException;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import com.oracle.truffle.api.dsl.CachedContext;
+import com.oracle.truffle.api.dsl.GenerateUncached;
+import com.oracle.truffle.api.dsl.ImportStatic;
+import com.oracle.truffle.api.dsl.NodeChild;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.object.DynamicObject;
+
+/** Casts a value into a BigInteger. */
+@GenerateUncached
+@ImportStatic(RubyGuards.class)
+@NodeChild(value = "value", type = RubyNode.class)
+public abstract class BigIntegerCastNode extends RubySourceNode {
+
+    public static BigIntegerCastNode create() {
+        return BigIntegerCastNodeGen.create(null);
+    }
+
+    public static BigIntegerCastNode create(RubyNode value) {
+        return BigIntegerCastNodeGen.create(value);
+    }
+
+    public abstract BigInteger executeCastBigInteger(Object value);
+
+    @Specialization
+    protected BigInteger doInt(int value) {
+        return BigInteger.valueOf(value);
+    }
+
+    @Specialization
+    protected BigInteger doLong(long value) {
+        return BigInteger.valueOf(value);
+    }
+
+    @Specialization(guards = "isRubyBignum(value)")
+    protected BigInteger doBignum(DynamicObject value) {
+        return Layouts.BIGNUM.getValue(value);
+    }
+
+    @Specialization(guards = "!isBasicInteger(value)")
```

Comment:
The guard should be `!isRubyInteger(value)` here so it excludes Bignum too (which is handled above).
In general it's best if all guards are exclusive. On the fast path (the generated `execute()` method), only guards of that `@Specialization` are checked, not other guards.